.docker/cypress/spec/cypress/integration/course_approval.spec.js

```diff
@@ -0,0 +1,120 @@
+import { register, edit } from '../pageHelpers/event.js'
+import { approve } from '../pageHelpers/pending_approvals.js'
+import { imitate, quitImitation } from '../pageHelpers/person.js'
+import { root_user, user_dicta as dicta, user_necessitatibus as necessitatibus, user_ipsam as ipsam,
+         bern_leitpfadikurs, empty_approval } from '../support/constants.js'
+
+describe('Course approvals', function () {
+
+  beforeEach(() => {
+		cy.login(root_user.username, root_user.password)
+	})
+
+  it('can be issued via the UI or a request, the result is the same', function () {
+    const QUERY = `begin
+      Event::Approval.joins('INNER JOIN event_participations ON event_participations.application_id = event_approvals.application_id')
+        .where('event_participations.id = ${dicta.event.participation_id}').as_json(except: :approved_at)
+    rescue StandardError => e
+      return e
+    end`
+
+    cy.getEventUrl(dicta.event.id).then((res) => {
+      cy.visit(`${res.url}/participations/${dicta.event.participation_id}`)
+    })
+    cy.contains('Freigeben').click()
+    cy.nearestInput('Tätigkeit').type('-')
+    cy.nearestInput('Stufe').type('-')
+    cy.nearestInput('Wie bewährt er/sie sich dabei? z.B. Einsatz, Qualität des/der TN').type('-')
+    cy.nearestInput('Wo liegen seine/ihre Stärken?').type('-')
+    cy.nearestInput('Wo sollte man ihn/sie fördern?').type('-')
+    cy.nearestInput('Wünsche, Anregungen oder Bemerkungen an die Kursleitung').type('-')
+    cy.contains('Freigeben').last().click()
+
+    cy.appEval(QUERY).as('json_ui')
+
+    cy.app('clean')
+    cy.app('start_transaction')
+
+    cy.login(root_user.username, root_user.password)
+
+    approve(dicta.event.id, dicta.event.participation_id, empty_approval)
+
+    cy.appEval(QUERY).then((res) => {
+      expect(this.json_ui).to.deep.equal(res)
+    })
+  })
+
+  it('turn the warning badge into a success badge in the applications overview', function () {
+    const full_name = `${dicta.last_name} ${dicta.first_name} / ${dicta.nickname}`
+
+    cy.getEventUrl(dicta.event.id).then((res) => {
+      cy.wrap(res.url).as('event_url')
+    })
+
+    // check the badge
+    cy.get('@event_url').then((url) => {
+      cy.visit(`${url}/application_market`)
+    })
+    cy.get(`tbody#participants tr:contains('${full_name}')`)
+      .find('td:eq(3) > span').should('have.attr','title', 'Kursfreigabe ausstehend')
```

Comment:
Hier checkst du ja nur das title-attribut des badges, nicht aber das eigentliche icon… Ist das gewollt so?
```suggestion
    cy.get(`tbody#participants tr:contains('${full_name}')`)
      .find('td:eq(3) > span').should('contain', '?').should('have.attr','title', 'Kursfreigabe ausstehend')
```